src/openforms/registrations/contrib/objects_api/api/views.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any
+
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import authentication, permissions, views
+
+from openforms.api.views import ListMixin
+
+from ..client import get_objecttypes_client
+from .serializers import ObjecttypeSerializer, ObjecttypeVersionSerializer
+
+
+@extend_schema_view(
+    get=extend_schema(
+        tags=["registration"],
+    ),
+)
+class ObjecttypesListView(ListMixin, views.APIView):
+    """
+    List the available Objecttypes.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = ObjecttypeSerializer
+
+    def get_objects(self) -> list[dict[str, Any]]:
+        with get_objecttypes_client() as client:
+            return client.list_objecttypes()
+
+
+@extend_schema_view(
+    get=extend_schema(
+        tags=["registration"],
+    ),
+)
+class ObjecttypeVersionsListView(ListMixin, views.APIView):
+    """
+    List the available versions for an Objecttype.
+    """
+
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = ObjecttypeVersionSerializer
+
+    def get_objects(self) -> list[dict[str, Any]]:
+        with get_objecttypes_client() as client:
+            return client.list_objecttype_versions(self.kwargs["submission_uuid"])
```

Comment:
I'm interested to see how this handles 404s for bogus UUIDs

Reply:
Objecttypes API raises a `ValidationError` but for some reason a 500 is returned. Should we validate on our side that it is a correct UUID?

Reply:
no, a 500 is never an intended/deliberate outcome. Create an issue in the objecttypes API for this case, and I think a 404 is more appropriate.

Reply:
Opened: https://github.com/maykinmedia/objects-api/issues/361